src/SimulationMap.cu

```diff
@@ -107,6 +124,19 @@ __device__ SimulationMap::~SimulationMap()
 }
 
 
+__device__ int SimulationMap::find_face_index(Face *face) const
+{
+    for(int i = 0; i < polyhedron->n_of_faces; ++i)
+    {
+        if(face == &polyhedron->faces[i])
+        {
+            return i;
+        }
+    }
+    return 0;
```

Comment:
Is there any reason not to return `-1` instead? If `0` is returned, from outside code it's impossible to determine whether a `Face` was found (and it's the very first one) or it was not found at all